src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -84,7 +84,17 @@ internal virtual BoundExpression BindSwitchExpressionCore(SwitchExpressionSyntax
 
         private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol operandType, bool hasErrors, DiagnosticBag diagnostics)
         {
-            // PROTOTYPE(patterns2): give an error if there is a bindable `_` in scope.
+            // give an error if there is a bindable `_` in scope.
+            var lookupResult = LookupResult.GetInstance();
+            var name = node.UnderscoreToken.ValueText;
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            this.LookupSymbolsWithFallback(lookupResult, name, arity: 0, useSiteDiagnostics: ref useSiteDiagnostics, options: LookupOptions.AllMethodsOnArityZero);
+            diagnostics.Add(node, useSiteDiagnostics);
+            if (lookupResult.IsMultiViable)
```

Comment:
Looking at `BindIdentifier` for comparison, should this be `if (lookupResult.Kind != LookupResultKind.Empty)` instead?

Reply:
No, I don't intend for inaccessible members to affect the result. I think that test is a mistake in `BindIdentifier` because of its affect on programs like this:

``` c#
class Base
{
    static int Main() => 0;
    private static int _ = 1;
}

class Derived: Base
{
    static void M1()
    {
        _ = M2(); // error!?
    }
    static int M2() => 2;
}
```

Reply:
Reported as https://github.com/dotnet/roslyn/issues/25798